File: BEmuJava/bemu/src/main/java/com/bloomberglp/blpapi/MarketElementInt.java
//------------------------------------------------------------------------------
// <copyright project="BEmu_maven" file="/BEmu_maven/bemu/src/main/java/com/bloomberglp/blpapi/MarketElementInt.java" company="Jordan Robinson">
//     Copyright (c) 2013 Jordan Robinson. All rights reserved.
//
//     The use of this software is governed by the Microsoft Public License
//     which is included with this distribution.
// </copyright>
//------------------------------------------------------------------------------

package com.bloomberglp.blpapi;

import com.bloomberglp.blpapi.Name;
import com.bloomberglp.blpapi.Schema;

public class MarketElementInt extends Element
{
	private final int _value;
	private final String _name;
	
	MarketElementInt(String name, int value)
	{
		this._name = name;
		this._value = value;
	}
	
    public Name name() throws Exception
    {
    	return new Name(this._name);
    }
    
    public int numValues()
    {
    	return 1;
    }
    
    public int numElements()
    {
    	return 0;
    }
    
    public boolean isComplexType()
    {
    	return false;
    }
    
    public boolean isArray()
    {
    	return false;
    }
    
    public boolean isNull()
    {
    	return false;
    }

    public Schema.Datatype datatype()
    {
    	return Schema.Datatype.INT32;
    }
    
    protected StringBuilder prettyPrint(int tabIndent) throws Exception
    {
    	return super.prettyPrintHelper(tabIndent, String.valueOf(this._value));
    }
    
    public Object getValue()
    {
    	return this._value;
    }
    
    public double getValueAsFloat64()
    {
        return this._value;
    }
    
    public int getValueAsInt32()
    {
        return this._value;
    }
    
    public long getValueAsInt64()
    {
        return this._value;
    }
    
    public float getValueAsFloat32()
    {
        return this._value;
    }
}

File: BEmu_cpp/src/MarketDataRequest/MarketElementInt.cpp
//------------------------------------------------------------------------------
// <copyright project="BEmu_cpp" file="src/MarketDataRequest/MarketElementInt.cpp" company="Jordan Robinson">
//     Copyright (c) 2013 Jordan Robinson. All rights reserved.
//
//     The use of this software is governed by the Microsoft Public License
//     which is included with this distribution.
// </copyright>
//------------------------------------------------------------------------------

#include "MarketDataRequest/MarketElementInt.h"

namespace BEmu
{
	namespace MarketDataRequest
	{
		MarketElementInt::MarketElementInt(const std::string& name, int value)
		{
			this->_name = name;
			this->_value = value;
		}

		Name MarketElementInt::name() const
		{
			return Name(this->_name.c_str());
		}

		size_t MarketElementInt::numValues() const
		{
			return 1;
		}

		size_t MarketElementInt::numElements() const
		{
			return 0;
		}

		bool MarketElementInt::isNull() const
		{
			return false;
		}

		bool MarketElementInt::isArray() const
		{
			return false;
		}

		bool MarketElementInt::isComplexType() const
		{
			return false;
		}

		int MarketElementInt::getValueAsInt32(int index) const
		{
			if(index == 0)
				return this->_value;
			else
				throw elementPtrEx;
		}

		long MarketElementInt::getValueAsInt64(int index) const
		{
			if(index == 0)
				return (long)this->_value;
			else
				throw elementPtrEx;
		}

		float MarketElementInt::getValueAsFloat32(int index) const
		{
			if(index == 0)
				return (float)this->_value;
			else
				throw elementPtrEx;
		}

		double MarketElementInt::getValueAsFloat64(int index) const
		{
			if(index == 0)
				return (double)this->_value;
			else
				throw elementPtrEx;
		}

		std::ostream& MarketElementInt::print(std::ostream& stream, int level, int spacesPerLevel) const
		{
			this->prettyPrintHelper(stream, level, spacesPerLevel, this->_value);
			return stream;
		}

	}
}